services/datamanager/builtin/builtin.go

```diff
@@ -237,27 +250,27 @@ func (svc *builtIn) initializeOrUpdateCollector(
 	if attributes.RemoteRobotName != "" {
 		remoteRobot, exists := svc.r.RemoteByName(attributes.RemoteRobotName)
 		if !exists {
-			return nil, errors.Errorf("failed to find remote %s", attributes.RemoteRobotName)
+			return nil, captureConfigsChanged, errors.Errorf("failed to find remote %s", attributes.RemoteRobotName)
 		}
 		res, err = remoteRobot.ResourceByName(resource.NameFromSubtype(resourceType, attributes.Name))
 	} else {
 		res, err = svc.r.ResourceByName(resource.NameFromSubtype(resourceType, attributes.Name))
 	}
 	if err != nil {
-		return nil, err
+		return nil, captureConfigsChanged, err
 	}
 
 	// Get collector constructor for the component subtype and method.
 	collectorConstructor := data.CollectorLookup(metadata)
 	if collectorConstructor == nil {
-		return nil, errors.Errorf("failed to find collector for %s", metadata)
+		return nil, captureConfigsChanged, errors.Errorf("failed to find collector for %s", metadata)
 	}
 
 	// Parameters to initialize collector.
 	interval := getDurationFromHz(attributes.CaptureFrequencyHz)
 	targetFile, err := datacapture.CreateDataCaptureFile(svc.captureDir, captureMetadata)
```

Comment:
captureMetadata is used to create a data capture file, but currently session ID is not used as part of the directory path, which is correct!

But where we actually use sessionID is `syncDataCaptureFiles()`, when you call the collector `attributes.SessionID` in order to build metadata that will be synced.
These collector attributes are taken directly from the config attributes passed into this method, so you would want to override `attributes.SessionID` before `svc.collectors[componentMetadata] = collectorAndConfig{collector, attributes}` below.